src/qibo/models/encodings.py

```diff
@@ -154,31 +180,91 @@ def unary_encoder_random_gaussian(nqubits: int, seed=None):
     return circuit
 
 
-def _generate_rbs_pairs(nqubits):
+def _generate_rbs_pairs(nqubits: int, architecture: str):
     """Generating list of indexes representing the RBS connections
-    and creating circuit with all RBS initialised with 0.0 phase."""
-    pairs_rbs = [[(0, int(nqubits / 2))]]
-    indexes = list(np.array(pairs_rbs).flatten())
-    for depth in range(2, int(math.log2(nqubits)) + 1):
-        pairs_rbs_per_depth = [
-            [(index, index + int(nqubits / 2**depth)) for index in indexes]
-        ]
-        pairs_rbs += pairs_rbs_per_depth
-        indexes = list(np.array(pairs_rbs_per_depth).flatten())
+
+    Creates circuit with all RBS initialised with 0.0 phase.
+
+    Args:
+        nqubits (int): number of qubits.
+        architecture(str, optional): circuit architecture used for the unary loader.
+            If ``diagonal``, uses a ladder-like structure.
+            If ``tree``, uses a binary-tree-based structure.
+            Defaults to ``tree``.
+
+    Returns:
+        (:class:`qibo.models.circuit.Circuit`, list): circuit composed of :class:`qibo.gates.gates.RBS`
+            and list of indexes of target qubits per depth.
+    """
+
+    if architecture == "diagonal":
+        pairs_rbs = np.arange(nqubits)
+        pairs_rbs = [[pair] for pair in zip(pairs_rbs[:-1], pairs_rbs[1:])]
+
+    if architecture == "tree":
+        pairs_rbs = [[(0, int(nqubits / 2))]]
+        indexes = list(np.array(pairs_rbs).flatten())
```

Comment:
```suggestion
        indexes = pairs_rbs[0][0]
```
this should be equivalent right?